fud2/fud-core/src/run.rs

```diff
@@ -147,13 +162,66 @@ impl EmitBuild for OpEmitData {
             }
         }
 
-        emitter.build_cmd_with_vars(
-            outputs,
-            &self.op_name,
-            inputs,
-            &[],
-            &vars,
-        )?;
+        // Some outputs are created by user specified `shell_deps`. Others are not and have to be
+        // created by running every `shell`/`shell_deps` command.
+        let mut uncreated_outputs: HashSet<String> =
+            HashSet::from_iter((0..outputs.len()).map(|i| {
+                format!("${}", crate::run::io_file_var_name(i, false))
+            }));
+
+        // Write a sequence of rules for each cmd.
+        for (i, cmd) in self.cmds.iter().enumerate() {
+            let rule_name = format!("{}_rule_{}", self.op_name, i + 1);
+            emitter.rule(&rule_name, &cmd.cmd)?;
+            let targets = cmd.gens.iter().map(|s| s.as_str()).collect_vec();
+            let deps = cmd.deps.iter().map(|s| s.as_str()).collect_vec();
```

Comment:
I think it continues to make sense to avoid `collect`ing these and to just emit them directly to the Ninja file with `emitter.arg`.

Reply:
I could be missing something important, but I think it would just look like using `build_cmd` as is, and then exiting these with `arg` afterward. But I really don't think this matters and probably shouldn't have brought it up in the first place; this is totally fine!